app/models/cms/group.rb

```diff
@@ -10,7 +10,7 @@ class CMS::Group < ApplicationRecord
   validates :name, :provider, presence: true, length: { maximum: 255 }
   validates :name, uniqueness: { scope: [:provider_id] }
 
-  has_many :group_sections, :class_name => 'CMS::GroupSection'
+  has_many :group_sections, :class_name => 'CMS::GroupSection', dependent: :delete_all
```

Comment:
why do you add this?

Reply:
> Also, I found that records in the cms_group_sections table are orphaned when deleting the group or the section, and added the :dependent options to the corresponding associations.

It's not directly related to the issue but I think it's necessary, maybe this would need another PR, that's why I created a draft.

Reply:
I know that many times these deps we left undefined to avoid issues when deleting stuff. And we have some delete object background workers. Whether here we have this or not I can't tell. But adding this will require some research.

Reply:
Then it'll be better to remove those lines from the PR and create a Jira issue specifically for this.